R/jm_complete.R

```diff
@@ -0,0 +1,39 @@
+
+#' @rdname jm_complete
+setMethod("jm_complete",
+    signature(Long = "LongModel", Os = "OsModel"),
+    value = "StanModule",
+    def = function(Long, Os) {
+        priors <- append(Long@stan@priors, Os@stan@priors)
+        priors <- priors[!duplicated(names(priors))]
+
+        StanModule(
+            functions = paste0(Long@stan@functions,
```

Comment:
I thought we have a general method to combine two stan objects together that would take care behind the scenes of combining all of this?

Reply:
This are specifically a `LongModel` and an `OsModel` though. They need to be treated in a special way compared to other cases.

Reply:
Interesting. In that case I would propose to write a method that works for these two - then the code is cleaner here / easier to read. This works because S4 can dispatch on multiple arguments, so when it sees exactly this combination then it will use the special method

Reply:
I am not sure I understand. I am using an S4 method (`setMethod`) with specific signatures for `LongModel` and `OsModel`.
https://github.com/Genentech/jmpost/blob/043e99a20f96307c6beb99d78292a9c86bc7746e/R/joint.R#L4
Is this different from your suggestion ?

Reply:
Maybe also an additional sidenote: `StanModule` is the core object that contains different parts of stan code. All objects contain it (`OsModel`, `LongModel` etc). I am referring to this case as "special" because it is different from the `parametrize` for example ( where we combine an `OsModel` with the link but practically the output is again a new `StanModule`).

Reply:
yeah it would be like here, but just to combine the two objects. probably I am just proposing to not define a new generic function `jm_complete()` but instead just define a method for the existing generic that combines two general stan objects.

Reply:
Daniel, you can now find the updated generic `merge` method including also what was previously known as `Joint` ( and even more previously as `jm_complete`). `merge` was initially constructed for merging object of identical class. I had to drop this restriction in order to include the merging of the longitudinal and overall survival model.

Reply:
thanks George, sounds good